src/crimson/os/seastore/random_block_manager/nvme_block_device.cc

```diff
@@ -192,43 +192,50 @@ write_ertr::future<> NVMeBlockDevice::writev(
     supported_stream = WRITE_LIFE_NOT_SET;
   }
   if (is_end_to_end_data_protection()) {
-    return seastar::do_with(
-      std::move(bl),
-      [this, offset] (auto &bl) {
-      return nvme_write(offset, bl.length(), bl.c_str());
-    });
+    co_await nvme_write(offset, bl.length(), bl.c_str());
+    co_return;
   }
   bl.rebuild_aligned(super.block_size);
+  auto iovs = bl.prepare_iovs();
+  std::vector<seastar::future<>> tasks;
+  tasks.reserve(iovs.size());
+
+  for (auto &p : iovs) {
+    auto off = offset + p.offset;
+    auto len = p.length;
+    auto iov = std::move(p.iov);
+    tasks.emplace_back(
+      seastar::futurize_invoke([this, off, supported_stream, len,
+      iov = std::move(iov)]() mutable -> seastar::future<> {
+      try {
+	size_t written = co_await io_device[supported_stream].dma_write(
+	  off, std::move(iov));
+	if (written != len) {
+	  logger().error("{} poffset={}~{} dma_write len={} inconsistent",
+	    device_id_printer_t{get_device_id()}, off, len, written);
+	  throw std::system_error(EIO, std::system_category(),
+	    "written len mismatch"); 
```

Comment:
We should probably make() write_ertr error here instead of throw. What do you think?
Can be done in a new commit since this one is only for coroutine switch.